internal/command/jsonformat/renderer.go

```diff
@@ -27,6 +30,27 @@ const (
 	CanNotApply
 )
 
+type JSONLogType string
+
+type JSONLog struct {
+	Message    string                `json:"@message"`
+	Type       JSONLogType           `json:"type"`
+	Diagnostic *viewsjson.Diagnostic `json:"diagnostic"`
+	Outputs    viewsjson.Outputs     `json:"outputs"`
+}
```

Comment:
This seems pragmatic for now.

For future: did you consider using something like mapstructure (or a similar technique) to inspect the type and decode into a detailed struct? This would allow us to more closely unify the UI output for local and remote operations, if both are rendering data from the same source. It would also allow us to more usefully colorize log messages where appropriate.

Reply:
I did not consider that. I was wondering if you could explain this in more detail -- how would mapstructure differ from `viewjson.Outputs` ?

Reply:
I am unfamiliar with mapstructure so I did not consider it. Is this in reference to the entire `JSONLog` struct or the `Outputs` field specifically? Wondering what advantages mapstructure would give us -- reading through the [README](https://github.com/mitchellh/mapstructure) at first glance seems like the library is a sensible option if we can't make any guarantees about the schema for logs.

Reply:
mapstructure is just an example of a technique for decoding differently-structured JSON messages into the appropriate structs. Looking through the Terraform code base, it seems like we don't tend to use mapstructure itself in modern code.

The advantage we'd get from doing this at all is that we would be able to decode different messages into different structures, which we could then render based upon that decoded data. While we could have a giga-struct which is the overlay of all possible structs, that's messy, confusing, and may not be possible if there are naming conflicts.

This isn't something we need to address now, by any means! As far as I can see, with this solution is a very reasonable tradeoff for now. I'd be happy to talk more or jam on this a bit as a follow-up, if you like.